<launch>

  <!-- Launch Kinect -->
	<include file="$(find perception3d)/launch/kinect.launch"></include> 
	
	<!-- Link Kinect to JACO -->
	<!--<node pkg="tf" type="static_transform_publisher" name="jaco_kinect" args="-0.60 0.385 0.54 0 0.30 0  arm_base camera_link 100" /> 
	<node pkg="tf" type="static_transform_publisher" name="jaco_kinect" args="-0.587 0.386 0.533 0 0.30 0  arm_base camera_link 100" /> 
	<node pkg="tf" type="static_transform_publisher" name="jaco_kinect" args="-0.61 0.40 0.52 -0.02 0.299 0.01178  arm_base camera_link 100" /> -->
<node pkg= "tf" type="static_transform_publisher" name="jaco_kinect" args=" -0.4777 0.5468 0.6775 -0.02 0.299 0.01178 arm_base camera_link 100" /> 
	
	<!-- Link AR_TAG to joint_6 (useful when calibrating) -->
	<!-- Disable for moveit implementation  Will put it back later on -->
	<node pkg="tf" type="static_transform_publisher" name="ARtag" args="0 0.045 -0.052 0 0 0 jaco_link_hand ARtag 100" /> 
	
	<!-- Plane Segmentation -->
  <node pkg="perception3d" type="segmentation" name="segmentation" respawn="false" output="screen">
    <param name="pcl_visualizer"  value="true" />
    <param name="topic_in" value="/camera/depth_registered/points" />
  </node>
  <!--
  <node pkg="perception3d" type="segmentation" name="segmentation" respawn="false" output="screen">
      <param name="pcl_visualizer"  value="true" />
      <param name="topic_in" value="/camera/depth_registered/points" />
      <param name="test" value="false" />
      <param name="file_path" value="/home/robot/rgbd-scenes-v2/pc/01.ply"/>
  </node> -->
	
	<!-- Euclidean Object Extraction -->
  <node pkg="perception3d" type="perception" name="perception" respawn="false" output="screen">
    <param name="objects_visualizer"  value="true" />
    <param name="directory_url" value="$(find perception3d)/database" />
    <param name="screenshot_url" value="$(find perception3d)/screenshots/" />
    <param name="planning_plugin" value="ompl_interface/OMPLPlanner" />
    <param name="request_adaters" value="default_planner_request_adapters" />
  </node>
	
  <node pkg="perception3d" type="image_transport_node" name="image_transport_node" respawn="false" output="screen"/>
	
</launch>


